src/features/session_trace/aggregate/index.js

```diff
@@ -1,538 +1,220 @@
+    if (firstSessionHarvest) this.agentRuntime.session.write({ traceHarvestStarted: true })
 
-      if (!(name in toAggregate)) return listOfSTNodes
-      // Special processing for event nodes dealing with user inputs:
-      const reindexByOriginFn = this.smearEvtsByOrigin(name)
-      const partitionListByOriginMap = listOfSTNodes.sort((a, b) => a.s - b.s).reduce(reindexByOriginFn, {})
-      return Object.values(partitionListByOriginMap).flat() // join the partitions back into 1-D, now ordered by origin then start time
-    }, this)
-    if (stns.length === 0) return {}
+    const hasReplay = this.agentRuntime.session?.state.sessionReplayMode === 1
+    const endUserId = this.agentInfo?.jsAttributes?.['enduser.id']
 
-    if (retry) {
-      this.sentTrace = this.trace
-    }
-    this.trace = {}
-    this.nodeCount = 0
-
-    let firstHarvestOfSession
-    if (this.agentRuntime.session) {
-      const isFirstPayload = !this.agentRuntime.session.state.traceHarvestStarted
-      firstHarvestOfSession = { fsh: Number(isFirstPayload) } // converted to '0' | '1'
-      if (isFirstPayload) this.agentRuntime.session.write({ traceHarvestStarted: true })
-    }
+    this.everHarvested = true
 
+    /** The blob consumer expects the following and will reject if not supplied:
+     * browser_monitoring_key
+     * type
+     * app_id
+     * protocol_version
+     * attributes
+     *
+     * For data that does not fit the schema of the above, it should be url-encoded and placed into `attributes`
+     */
     return {
       qs: {
-        st: this.agentRuntime.offset,
-        /** hr === "hasReplay" in NR1, standalone is always checked and processed before harvesting
-         * so a race condition between ST and SR states should not be a concern if implemented here */
-        hr: Number(!this.isStandalone),
-        /** fts === "firstTimestamp" in NR1, indicates what the earliest NODE timestamp was
-         * so that blob parsing doesn't need to happen to support UI/API functions  */
-        fts: this.agentRuntime.offset + earliestTimeStamp,
-        /** n === "nodeCount" in NR1, a count of nodes in the ST payload, so that blob parsing doesn't need to happen to support UI/API functions */
-        n: stns.length, // node count
-        ...firstHarvestOfSession
+        browser_monitoring_key: this.agentInfo.licenseKey,
+        type: 'BrowserSessionChunk',
+        app_id: this.agentInfo.applicationID,
+        protocol_version: '0',
+        attributes: encodeObj({
+          // this section of attributes must be controllable and stay below the query param padding limit -- see QUERY_PARAM_PADDING
+          // if not, data could be lost to truncation at time of sending, potentially breaking parsing / API behavior in NR1
+          // trace payload metadata
+          'trace.firstTimestamp': this.agentRuntime.offset + earliestTimeStamp,
+          'trace.lastTimestamp': this.agentRuntime.offset + latestTimeStamp,
```

Comment:
Can reduce length by sending the relative time and letting DEM do the calculation bc unix time is very long.

Reply:
Should also reduce some of these long names. Like `trace.` prefix is not necessary, and `start` under a trace chunk should be understandable to services working on it. In general, we've been very lax lately on name sizing--which is more of an issue for query params than payload bodies; idk if that's a good or bad thing as transfer sizes is both a cost to us and extra bill to the customer(?). I imagine that was the bigger reason things were packed or abbreviated in "the old days" (vs. any old browser concerns with payload size).

Reply:
there is no payload size concern here.  The blob consumer does not do any translating of data (ie magic) so it's important to provide the properties with a certain level of explicitness.  Not only this, but we are aligning to a cross team spec on these matters.  Accordingly, the timestamp values are not going to be evaluated by the backend for performance reasons, and the API needs raw searchable values for timestamps for querying and paging purposes.  I would suggest you take some time to read the rum specs, timestamp specs, and blob consumer specs if you have concerns here.